core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java

```diff
@@ -271,6 +273,22 @@ private InvalidEntry<StructuredRecord> getErrorObject(Map result, StructuredReco
   private Object decode(Object object, Schema schema) {
     Schema.Type type = schema.getType();
 
+    Schema.LogicalType logicalType = schema.getLogicalType();
+    if (logicalType != null) {
+      switch (logicalType) {
+        case DECIMAL:
+          BigDecimal bigDecimal = null;
+          if (object instanceof Double) {
```

Comment:
Can object also be `instanceof float`?

Reply:
I doubt, did not get a float while testing , js has all numbers 64 bits, won't fit in a float i think ...

Reply:
From [stack-overflow](https://stackoverflow.com/questions/25989642/why-does-java-8-nashorn-javascript-modulo-returns-0-0-double-instead-of-0-i) comments and docs i get it's only `Double or Integer`, I have even tested with int max, int max + 1 and JS native float 32. I got these two always.

Anyway as per  [docs](https://docs.oracle.com/javase/8/docs/technotes/guides/scripting/prog_guide/javascript.html#A1147390) under `3.10 Mapping Data Types` it's suggested we should expect the `Number` type. 

Adding this code snippet should cover all the cases !

```java
if (object instanceof Number) {
  double doubleValue = ((Number) object).doubleValue();
  bigDecimal = BigDecimal.valueOf(doubleValue).setScale(schema.getScale(), RoundingMode.HALF_EVEN);
 }
```